components/clarinet-deployments/src/lib.rs

```diff
@@ -452,15 +447,16 @@ pub async fn generate_default_deployment(
             // Extract the known / unknown dependencies
             match dependencies {
                 Ok(inferable_dependencies) => {
-                    // Looping could be confusing - in this case, we submitted a HashMap with one contract, so we have at most one
-                    // result in the `inferable_dependencies` map. We will just extract and keep the associated data (source, ast, deps).
-                    for (contract_id, dependencies) in inferable_dependencies.into_iter() {
+                    // We submitted a HashMap with one contract, so we have at most one result in the `inferable_dependencies` map.
+                    // We will extract and keep the associated data (source, ast, deps).
+                    if let Some((contract_id, dependencies)) =
+                        inferable_dependencies.into_iter().next()
+                    {
                         for dependency in dependencies.iter() {
                             queue.push_back((dependency.contract_id.clone(), None));
                         }
                         requirements_deps.insert(contract_id.clone(), dependencies);
                         requirements_data.insert(contract_id.clone(), (clarity_version, ast));
-                        break;
                     }
                 }
```

Comment:
@lgalabru Can you double check that I didn't break anything here?

Instead of doing only one iteration in a for loop, do `inferable_dependencies.into_iter().next()`

Reply:
The code seems equivalent yes. Should we log `warn` if `dependencies.len` > 1?

Reply:
Will do

Reply:
Or should we use `inferable_dependencies.first()` instead?

Reply:
`inferable_dependencies` is a BTreeMap and `BTreeMap::first_entry()` works a bit differently.
With `into_iter().next()` we stay very close to the initial implementation

Reply:
Added a log